### PR TITLE
Beyza GOK module2 js week3

### PR DESCRIPTION
**Pull request by beyzagoknl**

### CODE REVIEW

File: 1-JavaScript/Week3/homework/ex3-tellFortune.js [hunk 1]
```diff
@@ -33,31 +33,23 @@ body, this code is now written once only in a separated function.
...
-  // TODO complete this function
+function selectRandomly(randomArray) {
+  return randomArray[Math.floor(Math.random() * randomArray.length)];
 }
 
-function tellFortune(/* TODO add parameter(s) here */) {
-  // TODO complete this function
+function tellFortune(numKids, partnerNames, locations, jobTitles) {
+  return `You will be a ${selectRandomly(jobTitles)} in ${selectRandomly(locations)}, married to ${selectRandomly(partnerNames)} with ${selectRandomly(numKids)} kids.`
 }
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex3-tellFortune.js:42]: Perfect 🎉 !!

File: 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js [hunk 2]
```diff
@@ -16,9 +16,14 @@ it pure. Do the following:
...
-function addToShoppingCart(/* TODO parameter(s) go here */) {
-  // TODO complete this function
-}
+function addToShoppingCart(shopArr, groceryItem) {
+  const updatedShoppingCart=[...shopArr, groceryItem];
+    if(updatedShoppingCart.length > 3) {
+      return updatedShoppingCart.slice(updatedShoppingCart.length - 3);
+    } else {
+      return updatedShoppingCart;
+    }
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex5-shoppingCartPure.js:25]: Great solution! Something to consider: is `if/else` necessary? Or can you do without?

File: 1-JavaScript/Week3/homework/ex6-totalCost.js [hunk 3]
```diff
@@ -21,22 +21,30 @@ instead!
...
 function test1() {
   console.log('\nTest 1: calculateTotalPrice should take one parameter');
-  // TODO replace this comment with your code
+  console.assert(calculateTotalPrice.length===1);
 }
 
 function test2() {
   console.log('\nTest 2: return correct output when passed cartForParty');
-  // TODO replace this comment with your code
+  console.assert(calculateTotalPrice(cartForParty));
```
Comment (bnhoogendijk) [inline 1-JavaScript/Week3/homework/ex6-totalCost.js:47]: This test will probably pass 🙂. What is actually asserted here?

**Review by bnhoogendijk — APPROVED** [review]
You did an excellent job on this weeks homework, Beyza! You clearly demonstrate that you understand the concepts and that you can apply them. I left you comments / questions but there is no need to reply. Keep coding like this!

**Comment by beyzagoknl** [comment]
> You did an excellent job on this weeks homework, Beyza! You clearly demonstrate that you understand the concepts and that you can apply them. I left you comments / questions but there is no need to reply. Keep coding like this!

Thank you very much for both your time and your comment. It is both a pleasure and an honor to hear such nice comments from you.

**Pull request closed without merging**